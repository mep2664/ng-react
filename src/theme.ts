export const fontFamily = 'Arial, "Arial Header", sans-serif';

export interface IEmphasis {
    emphasis: emphasisType;
}

interface IEmphasisTypes {
    Primary: string;
    Secondary: string;
    Dark: string;
    Light: string;
    Positive?: string;
    Neutral?: string;
    Warning?: string;
    Error?: string;
    Overlay?: string;
    Header?: string;
}

export type emphasisType = "Primary" | "Secondary" | "Dark" | "Light" |
    "Positive" | "Neutral" | "Warning" | "Error" | "Overlay" | "Header";

export const bgColor: IEmphasisTypes = {
    Primary: "#2185d0",
    Secondary: "#e0e1e2",
    Dark: "rgb(25, 25, 25)",
    Light: "#f0f0f0",
    Positive: "#21ba45",
    Warning: "orange",
    Error: "#d8000c",
    Overlay: "rgba(0,0,0,0.1)",
    Header: "rgb(40, 40, 40, 0.96)",
}

export const borderColor: IEmphasisTypes = {
    Primary: "#ffffff",
    Secondary: "#000000",
    Dark: "#0f0f0f",
    Light: "#f0f0f0",
}

export const buttonBgColor: IEmphasisTypes = {
    Primary: "#2185d0",
    Secondary: "#e0e1e2",
    Dark: "#1b1c1d",
    Light: "#f0f0f0",
    Positive: "#21ba45",
    Warning: "orange",
    Error: "#d8000c",
}

export const fontColor: IEmphasisTypes = {
    Primary: "#f0f0f0",
    Secondary: "#1f1f1f",
    Dark: "#1f1f1f",
    Light: "#f0f0f0",
    Positive: "#f0f0f0",
}

export const hoverBgColor: IEmphasisTypes = {
    Primary: "#1a49d6",
    Secondary: "#a9a9a9",
    Dark: "#000000",
    Light: "#ffffff",
}

export const hoverFontColor: IEmphasisTypes = {
    Primary: "#1a49d6",
    Secondary: "#a9a9a9",
    Dark: "#000000",
    Light: "#ffffff",
}
