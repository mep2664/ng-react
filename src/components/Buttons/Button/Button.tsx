import React from "react";
import { emphasisType, backgroundColor, fontColor, hoverBackgroundColor } from "../../../theme";
import styled from "styled-components";

interface IStyledButton {
    Emphasis: emphasisType;
}

const StyledButton = styled.button<IStyledButton>`
    background-color: ${(props) => backgroundColor[props.Emphasis]};
    border: none;
    border-radius: 3px;
    color: ${(props) => fontColor[props.Emphasis]};
    cursor: pointer;
    font-size: 15px;
    padding: 10px;
    vertical-align: center;

    &:hover {
        background-color: ${(props) => hoverBackgroundColor[props.Emphasis]};
    }
`;

interface IButtonProps {
    Caption: string;
    Type: "button" | "submit" | "reset";
    Emphasis: emphasisType;

    OnClick?: (e: React.MouseEvent<HTMLButtonElement>) => void;

}

export const Button: React.FC<IButtonProps> = ({ Caption, Emphasis, Type, OnClick }) => {
    return (
        <StyledButton
            type={Type}
            Emphasis={Emphasis}
            onClick={OnClick}
        >
            {Caption}
        </StyledButton>
    );
}
