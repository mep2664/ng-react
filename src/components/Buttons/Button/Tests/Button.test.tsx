import React from "react"
import { render, fireEvent } from "@testing-library/react"
import "@testing-library/jest-dom/extend-expect";
import { emphasisType, backgroundColor, fontColor, hoverBackgroundColor } from "../../../../theme";
import { Button } from "../Button"
import { ButtonType } from "../../";

describe("Tests for the Button component", () => {
    const Caption = "BUTTON_CAPTION";
    const OnClick = jest.fn();
    const Emphasis: emphasisType = "Primary";
    const Type: ButtonType = "button";

    const props = {
        Caption, OnClick, Emphasis, Type,
    };

    test("Component matches snapshot", () => {
        const { container } = render(
            <Button
                {...props}
            />,
        );
        expect(container).toMatchSnapshot();
    });

    test("Button exists", () => {
        const { getByText } = render(
            <Button
                {...props}
            />,
        )
        expect(getByText(Caption)).toBeDefined();
    });

    test("Emphasis style is applied correctly", () => {
        const { container } = render(
            <Button
                {...props}
            />,
        )
        const button = container.firstElementChild as HTMLElement;
        expect(button).toHaveStyle(`background-color: ${backgroundColor[Emphasis]}; color: ${fontColor[Emphasis]}`);

        fireEvent.mouseOver(button);
        expect(button).toHaveStyle(`background-color: ${hoverBackgroundColor[Emphasis]}`);
    });

    test("onClick event fires correctly", () => {
        const { container } = render(
            <Button
                {...props}
            />,
        )
        const button = container.firstElementChild as HTMLButtonElement;
        fireEvent.click(button);
        expect(OnClick).toHaveBeenCalledTimes(1);
    });
});
