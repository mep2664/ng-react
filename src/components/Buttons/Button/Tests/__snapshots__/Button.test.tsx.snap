// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for the Button component Component matches snapshot 1`] = `
<div>
  <button
    class="sc-bdVaJa eVEMrn"
    type="button"
  >
    BUTTON_CAPTION
  </button>
</div>
`;
